infra/network-isolation.bicep

```diff
@@ -47,6 +54,9 @@ module vnet './core/networking/vnet.bicep' = if (usePrivateEndpoint) {
           addressPrefix: '10.0.3.0/24'
           privateEndpointNetworkPolicies: 'Enabled'
           privateLinkServiceNetworkPolicies: 'Enabled'
+          networkSecurityGroup: {
+            id: usePrivateEndpoint ? nsg.outputs.id : ''
```

Comment:
Hm, I think Sarah suggested to me that there should be a different NSG for each subnet. Did you consider that approach?